apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Filter.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+
+using Google.Cloud.Firestore.V1;
+using System;
+using System.Collections.Generic;
+using FieldOp = Google.Cloud.Firestore.V1.StructuredQuery.Types.FieldFilter.Types.Operator;
+
+namespace Google.Cloud.Firestore;
+
+
+/// <summary>
+/// 
+/// </summary>
+public class Filter
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    internal class UnaryFilter : Filter
+    {
+        public FieldPath Field { get; }
+
+        // The integer value of either the UnaryOp or FieldOp, depending on whether _value is null.
+        public FieldOp _op { get; }
+
+        // The value for a field (binary) operator, or null for a unary operator
+        public Object _value { get; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="fieldPath"></param>
+        /// <param name="op"></param>
+        /// <param name="value"></param>
+        public UnaryFilter(FieldPath fieldPath, FieldOp op, Object value)
+        {
+            Field = fieldPath;
+            _op = op;
+            _value = value;
+        }
+    }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    internal class CompositeFilter : Filter

```

Comment:
I suspect you'll want a "WithAdditionalFilter" method or similar in order to implement my suggestion of keeping this as the only filter field within Query.